### PR TITLE
Implement query-comment

### PR DESCRIPTION
**Pull request by damian3031**
resolves https://github.com/starburstdata/dbt-trino/issues/305
- implemented query comments
- fixed bad tests from dbt-tests-adapter on our side. Could be reverted when tests will be fixed on dbt-core side.

### CODE REVIEW

File: tests/functional/adapter/test_query_comments.py [hunk 1]
```diff
@@ -8,25 +10,39 @@
 )
 
 
+# TODO: below tests could be simplified to just
```
Comment (hovaesco) [inline tests/functional/adapter/test_query_comments.py:13]: have you checked for any open issues/PRs related to this issue?

Comment (damian3031) [reply tests/functional/adapter/test_query_comments.py:13]: Yes I checked, but I didn't find any such issue. I will open issue in dbt-core myself.

File: tests/functional/adapter/test_query_comments.py [hunk 2]
```diff
@@ -8,25 +10,39 @@
...
+# pass statements, when tests in dbt.tests.adapter
+# will be fixed
 class TestQueryCommentsTrino(BaseQueryComments):
-    pass
+    def test_matches_comment(self) -> bool:
+        logs = self.run_get_json()
+        assert r"/* dbt\nrules! */\n" in logs
```
[the remark this answers is not in the document]
Comment (damian3031) [reply tests/functional/adapter/test_query_comments.py:36]: good point, I also think that it is better than searching in logs. I added new assertions and left old ones, as this is a way it is implemented in `dbt-core`

**Review by hovaesco — APPROVED** [review]

File: dbt/adapters/trino/connections.py [hunk 3]
```diff
@@ -482,6 +482,8 @@ def add_query(self, sql, auto_begin=True, bindings=None, abridge_sql_log=False):
             if without_comments == "":
                 continue
 
+            individual_query = self._add_query_comment(individual_query)
```
Comment (mdesmet) [inline dbt/adapters/trino/connections.py:485]: How is this query comment persisted in the database? Trino only has client_tags and object comments?

Comment (damian3031) [reply dbt/adapters/trino/connections.py:485]: [query-comment](https://docs.getdbt.com/reference/project-configs/query-comment) is a string added as a comment at the beginning (or [end](https://docs.getdbt.com/reference/project-configs/query-comment#append-a-custom-comment)) of each query.

e.g. when using the following config:
```yaml
query-comment: "executed by dbt"
```
The resulting queries will have a comment prepended like this:
```sql
/* executed by dbt */

select ...
```

**Pull request merged**